[3.3.5] Naxxanar and Voltarus teleporters fixed

PolarCookie opened the pull request:
**Changes proposed:**
Naxxanar teleporters would trap both alliance and horde player.
Alliance players would be trapped at the top since you turn in the quest "Last Rites" at Thessarian before leaving, thus causing the teleporter which requires the quest in your log to stop working.
Horde players would be trapped just because their exit teleporter (5339) was not scripted.

Voltarus teleporters were incorrectly assigned; lower teleporter would go to the upper chamber while upper teleporter would exit from Voltarus.
Teleporters could be used freely without needing the scourge disguise.

**Target branch(es):** 
- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.)
Builds and works ingame

**Known issues and TODO list:**
- [x] Fix https://www.wowhead.com/spell=52240/drakuramas-teleport-02
- [x] Fix teleport delay

Commit:
Naxxanar and Voltarus teleporters fixed

Commit:
Naxxanar and Voltarus teleporters fixed

ZenoX92 committed:
DB/Quest: The Everstill Bridge

Closes #22099

ZenoX92 committed:
DB/Waypoints: Defias Messenger

Closes #22100

ZenoX92 committed:
DB/Quest: Return to Quinn

Closes #22101

ZenoX92 committed:
DB/Quest: Quest Emotes (Feralas) (Alliance)

Closes #22102

ZenoX92 committed:
DB/Quest: Quest Emotes (Winterspring)

Closes #22103

ZenoX92 committed:
DB/Instance: Blackrock Depths

Closes #22104

ZenoX92 committed:
DB/Gossip: Malyfous Darkhammer

Closes #22105

ZenoX92 committed:
DB/Gossip: Malyfous's Catalogue

Closes #22106

ZenoX92 committed:
DB/Gossip: Finkle Einhorn

Closes #22107

ZenoX92 committed:
DB/Spawns: Remove some duplicates

Closes #22108

ZenoX92 committed:
DB/Quest: Return to Vahlarriel

Closes #22112

ZenoX92 committed:
DB/Quest: Down the Scarlet Path (Part 2)

Closes #22113

Killyana committed:
Creature/Script: Remove deprecated script

Shauren committed:
Core/Players: Fixed saving quest item progress

Shauren committed:
SQL: Added missing updates entry

ZenoX92 committed:
DB/SAI: Dun Morogh

Closes #22139

ZenoX92 committed:
DB/Quest: Quest Emotes (Desolace) (Alliance)

Closes #22140

ZenoX92 committed:
DB/Gossip: Magistrate Bluntnose

Closes #22141

ZenoX92 committed:
DB/Gossip: Vyrin Swiftwind & Marek Ironheart

Closes #22142

ZenoX92 committed:
DB/SAI: Loch Modan

Closes #22143

illfated committed:
DB/Creature: Warosh, Bijou and Spirestone Mystic (LBRS)

Closes #22144

PolarCookie committed:
DB/quest: Engineering Membership Renewals

Closes #22134

PolarCookie committed:
DB/Loot: A Mangled Journal

Closes #22131

PolarCookie committed:
DB/Quest: Seared Scourge (daily)

Closes #22129

PolarCookie committed:
DB/Loot: Clam Meat/Pearl loot table adjustments

Closes #22128

Jonne733 committed:
DB/Quest: Skywing

Closes #22078
Closes #15228

Killyana committed:
DB: rename file

Commit:
Creature/Script: remove obsolete script npc_skywing (#22147)

- replaced with SAI script for the same quest and creature (Skywing)

Ref. https://github.com/TrinityCore/TrinityCore/issues/22078#issuecomment-399760597

Maphaze committed:
Core/GameObjects: Fixed DestroyGate battleground hook for vehicles (#22145)

ZenoX92 committed:
DB/Quest: Resupplying the Excavation

Closes #22148

ZenoX92 committed:
DB/Quest: A Dark Threat Looms (Part 7)

Closes #22149

ZenoX92 committed:
DB/Quest: Quest Emotes (Loch Modan)

Closes #22150

PolarCookie committed:
DB/Quest: Kurzen's Mystery

Closes #22130

Aokromes committed:
make linux happy

Keader committed:
Scripts/Black Temple: Removing double hook call in Spiritual Vengeance

(i realy dont remember why i did it lol)

Jildor committed:
Core/Spells: Shadow of Death & Spiritual Vengeance should be negative spells (BlackTemple/Teron Gorefiend) (#22164)

Closes: #22160
Closes: #22161

Treeston committed:
Core/PassiveAI: Fix an incorrect check that was inducing evade spam with critters. Closes #19998, for real this time.

Treeston committed:
Scripts/Commands: .npc info and .gobj info now show runtime AI typenames in addition to AIName/ScriptName.

Trying to figure out what base AI type a given creature is using has annoyed me one too many times (hi there, e8f9068).

Treeston committed:
Scripts/Commands: Demangled typeinfo of f7466c2 for gcc/cmake

Treeston committed:
Misc: Pass std::chrono types by value everywhere.

Treeston committed:
Core/GameObject: Fix an edge case where GOs might never respawn after being manually despawned. Closes #21792.

Killyana committed:
DB/SAI: Power Converter

Ref  #21986

ZenoX92 committed:
DB/Quest: Operation Recombobulation

Closes #22162

ZenoX92 committed:
DB/Creature: Mux Manascrambler

Closes #22167

ZenoX92 committed:
DB/Gossip: Anthion Harmon & Falrin Treeshaper

Closes #22168

Treeston committed:
Common/Containers: Eliminate a edge case that could potentially lead to object invalidation.

Treeston committed:
Core/Handlers: Spell handler no longer downranks otherwise-negative spells because they have a potentially-positive aura effect. Closes #22153.

killerwife committed:
Core/PacketIO: Fixed SMSG_GUILD_BANK_LIST packet payload for enchantments and sockets (#22177)

Credit for reversing goes to Laizerox

Shauren committed:
Core/Spells: Fixed Righteous Defense crash caused by modifying target list inside effect handler

* Already handled in a script

Closes #22158

ZenoX92 committed:
DB/Conditions: Kharan Mighthammer

Closes #22181

ZenoX92 committed:
DB/Object: Thaurissan Relic

Closes #22180

ZenoX92 committed:
DB/Gossip: Royal Historian Archesonus

Coses #22179

ZenoX92 committed:
DB/Creature: Stone Watcher of Norgannon

Closes #22183

Killyana committed:
Creature/Script: Add missing emote to Princess Moira Bronzebeard

Closes #22182

Killyana committed:
DB/Loot: Gaffer Jack

Closes #22175
Closes #21690

Jonne733 committed:
DB/Creature: Sharth Voldoun and Pathaleon the Calculator's Image

Closes #22094

ZenoX92 committed:
DB/Quest: Protecting the Shipment

Closes #17223

Treeston committed:
Core/Movement: Properly resume chase/follow movement if it was paused by a spell cast. Closes #21900.

Treeston committed:
Core/Maps: Continue updating creatures that own auras applied to players. Closes #22110.

Treeston committed:
visibility range check, fac4700 follow-up

Treeston committed:
Scripts/Pets: Some cleanup for Snake Trap. Tagging #21847, though this doesn't address it (yet).

Treeston committed:
Core/GameObject: Hunter traps now always put the Hunter in combat with the target when triggered. Closes #21847.

Treeston committed:
revert 5df7ffa, hack

DanVS committed:
Core/Entities: Ethereal Soul-Trader (#22050)

ZenoX92 committed:
DB/Text: Arikara

Updates #20946

ZenoX92 committed:
DB/Quest: Replacement Phial

Closes #22189

ZenoX92 committed:
DB/Quest: Restoring the Necklace

Closes #22190

ZenoX92 committed:
DB/SAI: Alcaz Island

Closes #22191

ZenoX92 committed:
DB/Quest: Quest Emotes (Dun Morogh)

Closes #22192

ZenoX92 committed:
DB/SAI: Wetlands

Closes #22195

Shauren committed:
Scripts/Icecrown Citadel: Fixed target selection for Vampiric Bite and Blood Mirror after threat rewrite

Closes #21537

Killyana committed:
DB/Conditions: Klaven Mortwake

Closes #21973

ZenoX92 committed:
DB/Quest: Lightforge Ingots

Closes #22196

Ovahlord committed:
Core/Spells: added a helper to reset proc cooldowns (#22137)

ZenoX92 committed:
DB/Gossip: Captain Stoutfist

Closes #22202

ZenoX92 committed:
DB/Quest: The Lost Fragments

Closes #22208

ZenoX92 committed:
DB/Gossip: Brumn Winterhoof

Closes #22206

ZenoX92 committed:
DB/Quest: Quest Emotes (Wetlands)

Closes #22207

ZenoX92 committed:
DB/Text: Inspector Tarem

Closes #22220

ZenoX92 committed:
DB/Gossip: Captain Nials, Skuerto & Navigator Mehran

Closes #22219

Treeston committed:
AI/SmartScripts: Warn on startup for ACTION_INVOKER_CAST without invoker (same as invoker targets)

dr-j committed:
DB/SAI: Uniting the Shattered Amulet (3.3.5)

Fix effects of http://www.wowhead.com/spell=12938/fel-curse when used for http://www.wowhead.com/quest=3627/uniting-the-shattered-amulet conditions for target of Razelikh the Defiler and Servent of Razelikh already exist in db but should also be usable on the other 3 above units

Treeston committed:
AI/SmartScripts: Replace SMART_ACTION_INVOKER_CAST with SMART_ACTION_SELF_CAST in action 85. SMART_ACTION_INVOKER_CAST moves to 134.

Killyana committed:
DB/SAI: Update Invoker cast action to 134

Ref https://github.com/TrinityCore/TrinityCore/commit/0df3cb5bcc8097070b1f215032d3914d6a577f01

Killyana committed:
DB/SAI: Update some SAI targets

Closes #22250

Killyana committed:
DB/SAI: Update target for whisper text

Target self for whisper text type cannot be used anymore
Closes #22249

Killyana committed:
DB/SAI: Fix 2 DB errors related to invoker

ZenoX92 committed:
DB/Gossip: Apprentice Kryten

Closes #22219

ZenoX92 committed:
DB/Quest: "Shady Rest Inn" Quest Chain (Horde)

Closes #22222

ZenoX92 committed:
DB/Texts: Smiling Jim

Closes #22230

ZenoX92 committed:
DB/Quest: Propaganda War

Closes #22231

ZenoX92 committed:
DB/Creature: Guard Lasiter

Closes #22232

ZenoX92 committed:
DB/Misc: Shady Rest Inn

Closes #22233

ZenoX92 committed:
DB/Object: The Park

Closes #22237

ZenoX92 committed:
DB/Texts: Acolyte Dellis

Closes #22238

ZenoX92 committed:
DB/Quest: The Missing Diplomat (Part 11)

Closes #22240

ZenoX92 committed:
DB/Quest: The Missing Diplomat (Part 12)

Closes #22241

ZenoX92 committed:
DB/Creature: Renn McGill

Closes #22242

ZenoX92 committed:
DB/Quest: Recover the Cargo!

Closes #22245

ZenoX92 committed:
DB/Quest: Trelane's Defenses

Closes #22246

Commit:
Scripts/Eastern Kingdoms: zone_wetlands converted to SAI cleanup (#22252)

Treeston committed:
Spells/Auras: Properly handle negative health modifiers on application. Closes #22211.

Treeston committed:
readability

Killyana committed:
DB/SAI: Update target relate to old template

Closes #22263

Treeston committed:
Scripts/Commands: Fix .quest complete to properly work with PvP kill quests. Closes #16321.

Killyana committed:
DB/Item: Lunar Festival Invitation

Closes  #11518

ZenoX92 committed:
DB/Quest: Cleansing Witch Hill

Closes #22262

ZenoX92 committed:
DB/Quest: The Missing Diplomat (Part 16)

Closes #22261

Killyana committed:
DB/Text: Maxima Blastenheimer

Closes #18338

Treeston committed:
Startup/DB: Print errors on npc_text vs broadcast_text mismatches.

Killyana committed:
DB: Typo

Killyana committed:
DB/Gossip: Selina Dourman

Closes #9961

Treeston committed:
Scritpts/Commands: Fix area name display if no area present. Closes #21727.

Killyana committed:
DB/Gossip: Vlagga Freyfeather and Amish Wildhammer

Ref #19398

Shauren committed:
Core/Commands: Fixed .damage command to work with spell schools caused by bad strtok usage

Closes #22280

Killyana committed:
DB/Creature: Fix some Npcs expansion on Skybreaker

Keader committed:
Scripts/Trial Of The Crusader: Fixed Anub'arak Pursuing Spikes

Closes #21420

ZenoX92 committed:
DB/Quest: "Shady Rest Inn" Quest Chain (Alliance)

Closes #22273

ZenoX92 committed:
DB/Quest: Disable some deprecated quests

Closes #22274

ZenoX92 committed:
DB/Quest: Quest Emotes (Arathi Highlands)

Closes #22275

ZenoX92 committed:
DB/Creature: Apprentice Morlann

Closes #22276

ZenoX92 committed:
DB/Creature: Stonemaul Spirit

Closes #22286

Killyana committed:
DB/Text: Spark Nilminer

illfated committed:
DB/Quest: Typos in RewardText; The Essence of Aku'Mai & King of the Foulweald

Closes #22289

Treeston committed:
Core/AI: EscortAI::SetRun now also adjusts already-in-progress waypoint paths, as opposed to only affecting future path loading. Closes #21290.

Treeston committed:
DBError follow-ups:

* broadcast_text mismatch (6eb7f06) is now a _INFO since Aok isn't nearly as fast as fixing as he claimed.
* SmartAI invoker misuse (482328b and a3c85c8) are now blocking errors

Treeston committed:
Core/SmartAI: SetRun fix for SAI, because for some reason it has like 95% of escortai code duplicated??? shitshow zz

closes #21463

Treeston committed:
Core/Movement: Prevent NO_PATH evades from incorrectly happening in a few places. Closes #22226.

Killyana committed:
DB/Quest: Suntara Stones

Closes #3305

Treeston committed:
Core/AI: Some cleanup to DoZoneInCombat and Malygos' AI that I found while working out #22226.

Killyana committed:
DB/Creature: Fix some npcs position in Eastern Plaguelands

Killyana committed:
DB/Spell: Flaming Spears

Ref #1596

Killyana committed:
DB/Condition: Plague Vial spell used only when player is in Taxi

Killyana committed:
DB/Creature: Steamrigger Mechanic spawns

DanVS committed:
Scripts/Scarlet Enclave: Death Comes From On High

Aokromes commented:
@PolarCookie what you mean with the re-use the bummy? it's needed to merge this?

Commit:
[3.3.5] Core/Loot/Gameobject: Tempspawned Gameobjects after looting (#22124)

* spell spawned chest non-despawn fix

* fix the consumed game objects

Killyana committed:
DB/Creature: Risen Wintergarde Miner

Closes #16306

Killyana committed:
DB: Typo

ZenoX92 committed:
DB/Creature: Stonemaul Spirit

Closes #22286

ZenoX92 committed:
DB/Quest: Move the Sunken Treasure to SAI

Closes #22290

ZenoX92 committed:
DB/Quest: Feast at the Blue Recluse

Closes #22291

Killyana committed:
DB/Creature: George Candarte

Closes #22293

offl committed:
DB/Locale: Zul'jin

Closes #22294

Killyana committed:
DB/GameObject: Restore deleted objects

jackpoz committed:
Core/Packets: Reduce memory footprint of cached queries (#22297)

Ensure only the minimum required memory is used by caching query packets by calling shrink_to_fit()

illfated commented:
"bummy"? is that a combined bunny + dummy? ;-) 😉

Treeston committed:
Core/LFG: Order group members before creating LFG group (leader > tank > healer > dps). Closes #16244.

Treeston committed:
Core/AI: Remove unnecessary parameter from DoZoneInCombat after eb1972f

Treeston committed:
Core/Creature: Some adjustments to Creature::CallForHelp. Now works even if the Creature hasn't selected a victim yet.

Treeston committed:
Scripts/SmartScripts: _SET_IN_COMBAT_WITH_ZONE and _CALL_FOR_HELP no longer use a target. Because that feature made no sense and was unused.

Treeston committed:
Core/Creature: Fall back to combat list if calling for help without threat list

Treeston committed:
Core/AI: Clean up charm AI handling, we now have two unique_ptr instead of a crapton of booleans

Treeston committed:
missing base class forward, 042f551 follow-up

Treeston committed:
wops (042f551 follow-up)

Treeston committed:
indent + header fix

PolarCookie commented:
@Aokromes, yes but wait.
Mechanically speaking this fix works; the Voltarus teleporters will work only if you have the disguise on, and scepter for the upper one, and the platforms have been corrected w.r.t. their destinations.
However the visual effect from the bummy (I like this word) is lost.

Since you asked, I've been looking into converting all this to SAI; I'll probably have it done by the weekend.

Treeston committed:
Core/Unit: Acquire GetControllingPlayer. Use to fix duel bugs.

Shauren committed:
Core/Creatures: Restored old initialization order: AI after movement

Closes #22307

Commit:
converted teleporters to SAI

Commit:
converted teleporters to SAI

Commit:
Update 9999_99_99_99_world.sql

escaped some

Commit:
Update 9999_99_99_99_world.sql

escaped some

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,81 @@
+(5340, 2, 0, 0, 46, 0, 100, 0, 5340, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 3687.91, 3577.28, 473.342,    0.0, "Areatrigger - On Trigger - Teleport player"),
+(5340, 2, 1, 0, 46, 0, 100, 0, 5340, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 3744.80, 3570.65, 341.66,     0.6, "Areatrigger - On Trigger - Teleport player"),
+(5339, 2, 0, 0, 46, 0, 100, 0, 5339, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 3733.68, 3563.25, 290.812, 3.6651, "Areatrigger - On Trigger - Teleport player"),
+(5338, 2, 0, 0, 46, 0, 100, 0, 5338, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 3733.68, 3563.25, 290.812, 3.6651, "Areatrigger - On Trigger - Teleport player");
```
Target type must be 8 if yo use coords.

PolarCookie replied:
I based it off 
```sql
SELECT * FROM smart_scripts WHERE entryorguid IN (5187, 5190) and source_type=2;
```
(5187, 2, 0, 0, 46, 0, 100, 0, 0, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 6150.76, -1071.38, 402.715, 2.1916, Areatrigger - On Trigger - Teleport)
(5190, 2, 0, 0, 46, 0, 100, 0, 0, 0, 0, 0, 0, 62, 571, 0, 0, 0, 0, 0, 7, 0, 0, 0, 6314.19, -1758.29, 457.071, 1.6787, Areatrigger - On Trigger - Teleport)
which is the K3- Garm's Rise teleporter pair.
target_type 7 works since I want to teleport the player but is this unintended mechanics?

Killyana replied:
Yes Keep the target 7, target 8 doesn't work in this case.

Killyana commented:
For the spell Drakuramas Teleport 02
``` sql
DELETE FROM `spell_target_position` WHERE `id` IN (52240);
INSERT INTO `spell_target_position` (`ID`, `EffectIndex`, `MapID`, `PositionX`, `PositionY`, `PositionZ`, `VerifiedBuild`) VALUES
(52240, 0, 571, 6175.59, -2000.67, 241.769, 0);

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,81 @@
+(-114829, 0, 0, 0, 38, 0, 100, 0, 1, 1, 0, 0, 0, 11, 52239, 0, 0, 0, 0, 0, 21, 10, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - On Data Set 1 1 - Cast 'Drakuramas Teleport Script 02'"),
+(-114830, 0, 0, 0, 38, 0, 100, 0, 1, 1, 0, 0, 0, 11, 52676, 0, 0, 0, 0, 0, 21, 10, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - On Data Set 1 1 - Cast 'Drakuramas Teleport Script 03'"),
+(-114831, 0, 0, 0, 38, 0, 100, 0, 1, 1, 0, 0, 0, 11, 52089, 0, 0, 0, 0, 0, 21, 10, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - On Data Set 1 1 - Cast 'Drakuramas Teleport Script 01'");
+
```
The teleport effect spell is linked with the teleport spell for 52089, 52676
``` sql
SELECT * FROM spell_linked_spell WHERE spell_trigger IN (52089, 52091, 52239, 52676, 52240,52677);

PolarCookie replied:
'Spell only hits player' as compared to hitting an npc?

Killyana replied:
Maybe the condition is wrong.

Killyana replied:
Yes it must hit the npc:
```
ServerToClient: SMSG_SPELL_START (0xDB2F) Length: 31 ConnIdx: 0 Time: 06/10/2011 16:57:29.000 Number: 110863
Caster GUID: Full: 0xF1306FC900000671 Type: Creature Entry: 28617 Low: 1649
Caster Unit GUID: Full: 0xF1306FC900000671 Type: Creature Entry: 28617 Low: 1649
Cast Count: 0
Spell ID: 52089 (52089)
Cast Flags: 10 (HasTrajectory, Unknown3)
Time: 0
Target Flags: 0 (Self)

PolarCookie replied:
That's a sniff, right?
What does it say about 52091?

Killyana replied:
Casted by the player on him self:
``` sql
ServerToClient: SMSG_SPELL_START (0xDB2F) Length: 29 ConnIdx: 0 Time: 06/10/2011 16:38:02.000 Number: 80387
Caster GUID: Full: xxxxxxxx Type: Player Low: xxxxxxx
Caster Unit GUID: Full: xxxxxxxxxxxx Type: Player Low: xxxxxx
Cast Count: 0
Spell ID: 52091 (52091)
Cast Flags: 2 (HasTrajectory)
Time: 0
Target Flags: 0 (Self)

Commit:
teleport spell fixes

Commit:
teleport spell fixes

PolarCookie commented:
@Killyana teleports are going up and down nicely, any good ideas about the trigger delay?

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world.sql:
```diff
@@ -0,0 +1,90 @@
+
+DELETE FROM `spell_linked_spell` WHERE `spell_trigger` IN (52239);
+INSERT INTO `spell_linked_spell` (`spell_trigger`, `spell_effect`, `type`, `comment`) VALUES
+(52239, 52240, 1, "Teleport");
```
Remove all this linked spells

PolarCookie replied:
and do the teleport manually through SAI?

Killyana replied:
Yes like this:
But I think the sai for https://woehead.way-of-elendil.fr/?npc=28948 must be updated as he use the visual spells linked to the teleport spell.
``` sql
DELETE FROM `smart_scripts` WHERE `source_type` = 2 AND `entryorguid` IN (5051, 5061, 5079, 5080);
DELETE FROM `smart_scripts` WHERE `source_type` = 9 AND `entryorguid` IN (505100, 506100, 507900, 508000);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(5051, 2, 0, 0, 46, 0, 100, 0, 5051, 0, 0, 0, 0, 80, 505100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5061, 2, 0, 0, 46, 0, 100, 0, 5061, 0, 0, 0, 0, 80, 506100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5079, 2, 0, 0, 46, 0, 100, 0, 5079, 0, 0, 0, 0, 80, 507900, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5080, 2, 0, 0, 46, 0, 100, 0, 5080, 0, 0, 0, 0, 80, 508000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(505100, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52089, 0, 0, 0, 0, 0, 19, 28617, 10, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast self 'Drakuramas Teleport Script 01"),
(505100, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52091, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast self 'Drakuramas Teleport 01'"),
(506100, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52239, 0, 0, 0, 0, 0, 19, 28617, 10, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 02"),
(506100, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52240, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list1 - Cast 'Drakuramas Teleport 02'"),
(507900, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52676, 0, 0, 0, 0, 0, 19, 28617, 10, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 03"),
(507900, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52677, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast 'Drakuramas Teleport 03'"),
(508000, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52089, 0, 0, 0, 0, 0, 19, 28617, 10, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 01"),
(508000, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52091, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast 'Drakuramas Teleport 01'");

PolarCookie replied:
I made something similar, but with the action list on the bunny, alas
```SmartAIMgr::LoadSmartAIFromDB: GUID-specific scripting not yet implemented for source_type 9```
I like this version since it solves a nagging suspicion I've been having with what happens when a lot of people approach the platform.

Killyana replied:
This SAI doesn't work for Two reason:
1- Trigger cannot run action list
2- If the target is closest npc the server crashs

Commit:
wip

Commit:
wip

PolarCookie commented:
Teleports arn't working anymore.

Killyana commented:
This version works, but only for 2, maybe because of conditions:
``` sql
UPDATE `creature_template` SET `AIName`="SmartAI", `ScriptName`="" WHERE `entry`=28617;
DELETE FROM `smart_scripts` WHERE `source_type` = 2 AND `entryorguid` IN (5051, 5061, 5079, 5080);
DELETE FROM `smart_scripts` WHERE `source_type` = 9 AND `entryorguid` IN (2861700, 2861701, 2861702, 2861703);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(5051, 2, 0, 1, 46, 0, 100, 0, 5051, 0, 0, 0, 0, 64, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Store target"),
(5051, 2, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 100, 1, 0, 0, 0, 0, 0, 10, 114832, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Send target"),
(5061, 2, 0, 1, 46, 0, 100, 0, 5061, 0, 0, 0, 0, 64, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Store target"),
(5061, 2, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 100, 1, 0, 0, 0, 0, 0, 10, 114829, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Send target"),
(5079, 2, 0, 1, 46, 0, 100, 0, 5079, 0, 0, 0, 0, 64, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Store target"),
(5079, 2, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 100, 1, 0, 0, 0, 0, 0, 10, 114830, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Send target"),
(5080, 2, 0, 1, 46, 0, 100, 0, 5080, 0, 0, 0, 0, 64, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Store target"),
(5080, 2, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 100, 1, 0, 0, 0, 0, 0, 10, 114831, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Send target"),
(5051, 2, 2, 0, 46, 0, 100, 0, 5051, 0, 0, 0, 0, 80, 2861700, 0, 0, 0, 0, 0, 10, 114832, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5061, 2, 2, 0, 46, 0, 100, 0, 5061, 0, 0, 0, 0, 80, 2861701, 0, 0, 0, 0, 0, 10, 114829, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5079, 2, 2, 0, 46, 0, 100, 0, 5079, 0, 0, 0, 0, 80, 2861702, 0, 0, 0, 0, 0, 10, 114830, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(5080, 2, 2, 0, 46, 0, 100, 0, 5080, 0, 0, 0, 0, 80, 2861703, 0, 0, 0, 0, 0, 10, 114831, 28617, 0, 0, 0, 0, 0, "Areatrigger - On Trigger - Action list"),
(2861700, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52089, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast self 'Drakuramas Teleport Script 01"),
(2861700, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52091, 0, 0, 0, 0, 0, 12, 1, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast self 'Drakuramas Teleport 01'"),
(2861701, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52239, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 02"),
(2861701, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52240, 0, 0, 0, 0, 0, 12, 1, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list1 - Cast 'Drakuramas Teleport 02'"),
(2861702, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52676, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 03"),
(2861702, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52677, 0, 0, 0, 0, 0, 12, 1, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast 'Drakuramas Teleport 03'"),
(2861703, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 85, 52089, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, "Areatrigger - Action list - Cast 'Drakuramas Teleport Script 01"),
(2861703, 9, 1, 0, 0, 0, 100, 0, 1000, 1000, 0, 0, 0, 85, 52091, 0, 0, 0, 0, 0, 12, 1, 0, 0, 0, 0, 0, 0, "Drakuramas Teleport Bunny 01 - Action list - Cast 'Drakuramas Teleport 01'");

Treeston committed:
Core/Unit: Store charmer and charmed pointers on Unit directly, no more map lookups for Unit::GetCharmer and Unit::GetCharmed

Killyana commented:
For the npc Malmortis no need to change anything on his SAI as he use the spell 52676 as cosmetic before he despawn.

Commit:
wip

Commit:
wip

Commit:
Merge branch 'Voltarus' of https://github.com/PolarCookie/TrinityCore into Voltarus

illfated commented:
Looks like you need to do a merge instead of rebase to align your branch `PolarCookie:Voltarus` with the `TrinityCore:3.3.5` branch.
Then again, that update could in theory mess up your own changes.

Aokromes commented:
i suggest you to close this pr and make a ticket instead, when the commit is pushed we will remove c++ part.
